kernel/src/actions/mod.rs

```diff
@@ -149,22 +157,21 @@ impl Protocol {
 #[derive(Debug, Clone, PartialEq, Eq, Schema)]
 #[cfg_attr(feature = "developer-visibility", visibility::make(pub))]
 #[cfg_attr(not(feature = "developer-visibility"), visibility::make(pub(crate)))]
-// TODO need to have a way to always write some fields but not always read them
 struct CommitInfo {
     /// The time this logical file was created, as milliseconds since the epoch.
-    /// TODO should this be a Timestamp?
-    pub(crate) timestamp: i64,
+    pub(crate) timestamp: Option<i64>,
     /// An arbitrary string that identifies the operation associated with this commit. This is
     /// specified by the engine.
-    pub(crate) operation: String,
+    pub(crate) operation: Option<String>,
```

Comment:
is this actually optional? Would a commit ever not have an associated action?

Reply:
it is indeed optional according to the protocol. that is, we should never _require_ it to be present when reading commit info

Reply:
right, to be clear I mean that it is optional because the protocol says commit info can be anything.

since we don't really have a way of communicating reader/writer optionality I can just add a comment saying that this is optional since it is _actually_ optional from a read/enforcement perspective but that kernel always writes it